poseidon2/io.nim

```diff
@@ -27,15 +35,17 @@ iterator elements*(bytes: openArray[byte], _: type F): F =
   ## sequences that end with 0x0's.
   const chunkLen = 31
   const endMarker = @[1'u8]
+  let empty = bytes.len == 0
   var chunkStart = 0
   while chunkStart + chunkLen <= bytes.len:
     let chunkEnd = chunkStart + chunkLen - 1
     let element = F.fromOpenArray(bytes.toOpenArray(chunkStart, chunkEnd))
     yield element
     chunkStart += chunkLen
-  let finalChunk = bytes[chunkStart..<bytes.len] & endMarker
-  let finalElement = F.fromOpenArray(finalChunk)
-  yield finalElement
+  if (bytes.len - chunkStart > 0) or empty:
```

Comment:
I think this was pulled from my padding PR which ended up being the wrong fix. Did something change, or was this pulled in by mistake?